### PR TITLE
Start Composing Revelio

### PR DESCRIPTION
**Pull request by willwill96**

### CODE REVIEW

File: library.tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (willwill96) [inline library.tsconfig.json]: This should inherit from the project tsconfig file

File: yarn.lock [hunk 2]
```diff
@@ -14278,8 +14328,9 @@ typescript@3.6.2:
   resolved "https://registry.yarnpkg.com/typescript/-/typescript-3.6.2.tgz#105b0f1934119dde543ac8eb71af3a91009efe54"
 
 typescript@^3.6.4:
-  version "3.6.4"
-  resolved "https://registry.yarnpkg.com/typescript/-/typescript-3.6.4.tgz#b18752bb3792bc1a0281335f7f6ebf1bbfc5b91d"
+  version "3.8.3"
```
Comment (willwill96) [inline yarn.lock]: upgraded typescript to support `declaration` and `allowJs` options at the same time https://github.com/microsoft/TypeScript/issues/7546

File: package.json [hunk 3]
```diff
@@ -4,11 +4,11 @@
   "main": "src/main/webapp/main.js",
   "license": "MIT",
   "devDependencies": {
-    "@connexta/ace": "https://github.com/connexta/ace.git#81e7b8b22b3f7e43b356ba5dfbc065ffdb694892",
+    "@connexta/ace": "https://github.com/willwill96/ace.git#7945d432e6d90fc587f234460a03983097ae7f6a",
     "@types/ol": "^5.3.6",
     "@types/react": "^16.9.5",
     "@types/react-loadable": "^5.5.3",
-    "@typescript-eslint/parser": "^2.3.3",
+    "@typescript-eslint/parser": "^2.22.0",
```
Comment (willwill96) [inline package.json]: needed to bump this version to support new typescript version

File: src/main/webapp/components/maps/draw-menu.tsx [hunk 4]
```diff
@@ -20,8 +20,9 @@ import {
 import { GeometryJSON } from 'geospatialdraw/bin/geometry/geometry'
 import DrawingToolbox from 'geospatialdraw/bin/drawing/drawing-toolbox'
 import useDrawingMenu from 'geospatialdraw/bin/menu/react-hooks'
+import { BoxProps } from '@material-ui/core/Box';
 
-type Props = React.HTMLProps<HTMLDivElement> & {
+type Props = BoxProps & {
```
Comment (willwill96) [inline src/main/webapp/components/maps/draw-menu.tsx]: This errored with the new typescript version

File: src/main/webapp/routes.js [hunk 5]
```diff
@@ -112,7 +112,7 @@ const loadDynamicRoute = route => {
     }),
     about: loadable({
       loader: () =>
-        import(/* webpackChunkName: "about" */ './components/about'),
+        import(/* webpackChunkName: "about" */ 'a-test-of-revelio/components/about'),
```
Comment (willwill96) [inline src/main/webapp/routes.js]: remove this

File: package.json [hunk 6]
```diff
@@ -1,29 +1,74 @@
 {
-  "name": "revelio",
-  "version": "1.0.0",
+  "name": "@connexta/revelio",
+  "version": "0.0.1",
   "main": "src/main/webapp/main.js",
   "license": "MIT",
   "devDependencies": {
-    "@connexta/ace": "https://github.com/connexta/ace.git#81e7b8b22b3f7e43b356ba5dfbc065ffdb694892",
+    "@apollo/react-hooks": "^3.1.3",
```
Comment (willwill96) [inline package.json]: Moved all dependencies to dev and peer dependencies so downstream project has to have the dependency. Maybe this is a bad idea?

File: src/main/webapp/components/query-schedule/query-schedule.tsx [hunk 7]
```diff
@@ -40,7 +40,7 @@ type QueryScheduleProps = {
 }
 
 const DatePicker = (props: KeyboardDatePickerProps) => {
-  const [inputValue, setInputValue] = useState()
+  const [inputValue, setInputValue] = useState<string | undefined>()
```
Comment (willwill96) [inline src/main/webapp/components/query-schedule/query-schedule.tsx]: This errored with new typescript version

File: package.json (hunk 6 above)
Comment (willwill96) [reply package.json]: With these in `dependencies` the yarn.lock when importing looks like this:
```
+a-test-of-revelio@1.0.47:
+  version "1.0.47"
+  resolved "https://registry.yarnpkg.com/a-test-of-revelio/-/a-test-of-revelio-1.0.47.tgz#de3902a70813342fd9fe0f1434bc243bfd3f2362"
+  integrity sha512-MGhFVegnDCzTMp15AFL/vFSL/FiZnIqqDzrcF5pYsVeHqo/3E0az4Hl+JxZD8MyOWxQrkcSzg5al+cMOAgya0Q==
+  dependencies:
+    "@apollo/react-hooks" "^3.1.3"
+    "@apollo/react-ssr" "^3.1.3"
+    "@date-io/moment" "1.3.13"
+    "@material-ui/core" "^4.6.1"
+    "@material-ui/icons" "^4.2.1"
+    "@material-ui/lab" "^4.0.0-alpha.32"
+    "@material-ui/pickers" "^3.2.5"
+    a-test-of-revelio "^1.0.40"
+    abort-controller "^3.0.0"
+    apollo-cache-inmemory "^1.6.3"
+    apollo-client "^2.6.4"
+    apollo-link "^1.2.13"
+    apollo-link-batch-http "^1.2.13"
+    apollo-link-error "^1.1.12"
+    apollo-link-http "^1.5.16"
+    apollo-link-schema "^1.2.4"
+    apollo-server-express "^2.9.7"
+    compression "^1.7.4"
+    express "4.17.1"
+    geospatialdraw "0.5.5"
+    golden-layout "^1.5.9"
+    graphql "^14.5.8"
+    graphql-tag "^2.10.1"
+    graphql-tools "^4.0.5"
+    immutable "^4.0.0-rc.12"
+    lodash.throttle "^4.1.1"
+    moment "^2.24.0"
+    moment-timezone "^0.5.16"
+    ol "^6.0.1"
+    on-finished "^2.3.0"
+    polished "^3.4.1"
+    qs "^6.9.1"
+    react "^16.9.0"
+    react-dom "^16.9.0"
+    react-loadable "^5.5.0"
+    react-redux "^7.1.0"
+    react-router "^5.1.2"
+    react-router-dom "^5.1.2"
+    redux "^4.0.4"
+    redux-thunk "^2.3.0"
+    styled-components "^4.3.2"
+    universal-cookie "^4.0.2"
+    uuid "^3.3.3"
+    winston "^3.2.1"

+a-test-of-revelio@^1.0.40:
+  version "1.0.48"
+  resolved "https://registry.yarnpkg.com/a-test-of-revelio/-/a-test-of-revelio-1.0.48.tgz#178d3fd46b130288749329dd8fcca594da3793e6"
+  integrity sha512-08Tme3qmHN2XxERoL0IImio26ms0NwVWm5F97HNVsmSMorpQPz/cAf0uq/4ns6DyCN7zqfQOVVSNz8nOm0/Aow==
```

With them moved to `peer/devDependencies`:
```
 
+a-test-of-revelio@^1.0.49:
+  version "1.0.49"
+  resolved "https://registry.yarnpkg.com/a-test-of-revelio/-/a-test-of-revelio-1.0.49.tgz#054eb58e7758ba33c3ed734e041b2417d3d99360"
+  integrity sha512-3i5B+bIhTwClSyCs9NZs9bkgexIjlWhPnzr84hz4oyMFmBabaCXng5G7IzIOpbZ/Tvo+tRDdR8n6oMJo2Nhc1w==

```

**Comment by andrewkfiedler** [comment]
👍 I'll try this out on a downstream project and let you know if I run into any issues.

**Pull request merged**